src/Microsoft.Data/DataFrame.cs

```diff
@@ -423,8 +420,8 @@ public GroupBy GroupBy(string columnName)
             return column.GroupBy(columnIndex, this);
         }
 
-        // In a GroupBy call, columns get resized. We need to set the RowCount to reflect the true Length of the DataFrame. Internal only. Should not be exposed
-        internal void SetTableRowCount(long rowCount)
+        // In GroupBy and ReadCsv calls, columns get resized. We need to set the RowCount to reflect the true Length of the DataFrame. This does internal validation
+        public void SetTableRowCount(long rowCount)
```

Comment:
Why are we exposing this now?

Reply:
Because when we read in the csv file, each `BaseColumn` appends values one by one. The columns don't have a reference to the `DataFrame` they are a part of though, so `DataFrame.RowCount` will still return 0. That's why I had to expose this method. I know this is not an ideal design, but I don't have a much better alternative? The `SetTableRowCount` method will do internal validation though and throw, so randomly setting a `RowCount` will fail for example. This method will only set the `RowCount` if all the `BaseColumn`s have the same `RowCount`.

Reply:
What if we built up all the columns (or even buffers) first, and then when we were done create a DataFrame around them?

Reply:
BTW - don't let this block you. You can log an issue and follow up on it later, if you want.